lib/msf/ui/console/command_dispatcher/developer.rb

```diff
@@ -217,6 +220,13 @@ def cmd_reload_lib_tabs(str, words)
     tab_complete_filenames(str, words)
   end
 
+  def cmd_reload_diff_files
+    files = %x(git diff --name-only).split
```

Comment:
can you add some error handling so that if git isn't available it fails nicely?

Reply:
I tested this and it already passes the error through to stdout. Did you have something else in mind?

Reply:
Error handling has been added. I had an idea that was better than my first attempt

Reply:
I would think to take this a step further and detect if source is even a git checkout.  Consider omnibus installations do not require git and also will not have a `.git` folder when to use when `git` is available.  We should be more friendly and catch both scenarios.

Ideas on messages, I understand these are similar to what the error messages that would already bubble up look like.
* no git binary "`git` is not found."
* no `.git` in console location "Framework installation is not a git repository"

"Did you install using a nightly installer or package manager?"

Reply:
`msfupdate` is probably a bad example, but we have done this before:

https://github.com/rapid7/metasploit-framework/blob/dcb96c20a01b38ce3ccd098251887ee42e42ee96/msfupdate#L117-L119

Reply:
If Git is unavailable, I'd rather it print out a message saying to install it than fail silently. Even if the directory is a Git repo, you might not have the command. I'd say check both.

Reply:
If Git is unavailable, I'd rather it print out a message saying to install it than fail with an unrelated error. Even if the directory is a Git repo, you might not have the command. I'd say check both.